lib/schema_tools/modules/hash.rb

```diff
@@ -192,6 +189,16 @@ def parse_object(obj, field, prop, opts)
         res
       end
 
+      private
+
+      def parse_object?(obj, field)
+        if obj.respond_to?( field )
+          rel_obj = obj.send( field )
```

Comment:
I thought about using `try` here as well, but [try](http://api.rubyonrails.org/classes/Object.html) actually built into rails --not ruby, so we probably don't want to use it here as it would force a rails dependency on this gem.